packages/nimble-components/src/table/specs/table-column-specs/table-column-anchor-hld.md

```diff
@@ -0,0 +1,173 @@
+    cellRecord: TCellRecord;
+    columnConfig: TColumnConfig;
+    columnUniqueId: string | undefined;
+}
+```
+
+This value we use for this unique ID could be the `columnId`, but that is `undefined` unless a client sets it. We would need to start auto-generating unique `columnId` values if the client hasn't provided one, or else document that users must provide column ids when wishing to integrate with the Angular router. Another issue with using `columnId` is that the client may change it programmatically, invalidating the value stored in the cell state. To ensure consistency, we would have to have a `columnIdChanged()` handler that caused `refreshRows()` to be called on the table.
+
+A better option would be to use the existing `internalUniqueId` that each column defines. These IDs are used for a couple purposes already where a unique, internal ID is needed.
+
+`columnUniqueId` is at the top level rather than part of the `columnConfig`, because it applies for all column types, and it is not for configuration.
+Adding `columnUniqueId` to the cell state would be a change to `nimble-components` just to support `nimble-angular`, but it seems generic enough to be a reasonable API change.
+
+Now that we have the column's `internalUniqueId` for a given anchor click, we need to get the column directive. It's possible that in the future, there will be other column types that contain `nimble-anchors`. For that reason, we'll have a base class for our directive that any anchor-containing column directive can extend.
+
+```ts
+abstract class NimbleTableColumnContainsAnchorDirective extends RouterLinkWithHref { ... }
+
+@Directive({ selector: 'nimble-table-column-anchor[nimbleRouterLink]' })
+export class NimbleTableColumnAnchorRouterLinkWithHrefDirective extends NimbleTableColumnContainsAnchorDirective { ... }
+```
+
+We can get all potential directives as content children:
+
+```ts
+@ContentChildren(NimbleTableColumnContainsAnchorDirective) public anchorColumnDirectives: QueryList<NimbleTableColumnContainsAnchorDirective>;
+```
+
+Each directive has a reference to its associated element, so we can get to the column element. From there, we have access to the `internalUniqueId`, so we can find the match.
+
+Finally, we need to call `RouterLinkWithHref.onClick()` to do the router navigation. However, we'll first need to set up the `routerLink` property with the url:
+
+```ts
+public doRouterNavigation(href: string): boolean {
+    this.routerLink = href;
+    return super.onClick(...);
+}
+```
+
+Note that `RouterLink`/`RouterLinkWithHref` [only handles left-clicks, when no other keys are held down](https://github.com/angular/angular/blob/9bd9a11f4e21e5a7cc9da18f150f6dd520e7cd1e/packages/router/src/directives/router_link.ts#L302). We will check the event state in our table directive, only calling `NimbleTableColumnContainsAnchorDirective.doRouterNavigation()` if it is a plain left-click. Otherwise, we let the default anchor action occur.
```

Comment:
Will the router link handle navigation by pressing the Enter key? I think it should, but the way this is worded makes me think it might not.

Reply:
Sorry, yes. The native handling of an Enter key on a link triggers a `click` event to be fired. So everything I said about handling plain left-clicks also applies to pressing Enter.